### PR TITLE
Fix some extract/restore issues.

### LINKED ISSUE #4661: restore issue: error creating commit: parent commit xxxxxxx has not been finished
**What happened?**:
Created a backup with --no-objects
Trying to restore, it is reporting error:
```
error creating commit: parent commit a99f85b89bd64df186dc207e7fd20de3 has not been finished
WARNING: Your cluster might be in an invalid state--consider deleting partially-restored data before continuing
```

Checking `pachd` logs:
```
2020-02-28T18:14:09Z INFO auth.API.WhoAmI {"request":{}} []
2020-02-28T18:14:09Z INFO pfs.API.BuildCommit {"duration":0.009365074,"request":{"parent":{"repo":{"name":"extraction"},"id":"1959138c183f4cd4b3df36ca6013759f"},"provenance":[{"commit":{"repo":{"name":"__spec__"},"id":"ced850264e9a4c6b9f40e5bf7358411c"},"branch":{"repo":{
"name":"__spec__"},"name":"extraction"}},{"commit":{"repo":{"name":"extraction_morning"},"id":"187d036de83042358e6654bf80ff86c2"},"branch":{"repo":{"name":"extraction_morning"},"name":"master"}}],"trees":[{"hash":"a8fad4504073983cd7bd64d91a9ae66e5f72fe75ccabb030d65150d533
04b327015c4546e18c7c5238c7a746dabe95ddd0385193e8f8d025a04c650446bc8c98"}],"datums":{"hash":"779706416e13ab92c0c9eb87d21e65d2af547a16461a1c1153567bc208af2a9a061f23620fa718a3f3ebf0ce6bd0abc3884cdb45d28726a1954a629d1fa18b42"},"ID":"b26887a935484b718bc82004872a2fc3","size_byt
es":1781132730},"response":{"repo":{"name":"extraction"},"id":"b26887a935484b718bc82004872a2fc3"}} []
2020-02-28T18:14:09Z INFO pfs.API.BuildCommit {"request":{"parent":{"repo":{"name":"extraction_exporter"},"id":"a99f85b89bd64df186dc207e7fd20de3"},"provenance":[{"commit":{"repo":{"name":"__spec__"},"id":"ced850264e9a4c6b9f40e5bf7358411c"},"branch":{"repo":{"name":"__spec
__"},"name":"extraction"}},{"commit":{"repo":{"name":"__spec__"},"id":"b400e6df43aa408ca43d68bf3cafd1ce"},"branch":{"repo":{"name":"__spec__"},"name":"extraction_exporter"}},{"commit":{"repo":{"name":"extraction"},"id":"b26887a935484b718bc82004872a2fc3"},"branch":{"repo":
{"name":"extraction"},"name":"master"}},{"commit":{"repo":{"name":"extraction_morning"},"id":"187d036de83042358e6654bf80ff86c2"},"branch":{"repo":{"name":"extraction_morning"},"name":"master"}}],"ID":"f7a48dde6aae4ddfb9374a54059cbbae"}} []
2020-02-28T18:14:09Z INFO auth.API.WhoAmI {"request":{}} []
2020-02-28T18:14:09Z ERROR pfs.API.BuildCommit {"duration":0.005019861,"error":"parent commit a99f85b89bd64df186dc207e7fd20de3 has not been finished","request":{"parent":{"repo":{"name":"extraction_exporter"},"id":"a99f85b89bd64df186dc207e7fd20de3"},"provenance":[{"commit
":{"repo":{"name":"__spec__"},"id":"ced850264e9a4c6b9f40e5bf7358411c"},"branch":{"repo":{"name":"__spec__"},"name":"extraction"}},{"commit":{"repo":{"name":"__spec__"},"id":"b400e6df43aa408ca43d68bf3cafd1ce"},"branch":{"repo":{"name":"__spec__"},"name":"extraction_exporte
r"}},{"commit":{"repo":{"name":"extraction"},"id":"b26887a935484b718bc82004872a2fc3"},"branch":{"repo":{"name":"extraction"},"name":"master"}},{"commit":{"repo":{"name":"extraction_morning"},"id":"187d036de83042358e6654bf80ff86c2"},"branch":{"repo":{"name":"extraction_mor
ning"},"name":"master"}}],"ID":"f7a48dde6aae4ddfb9374a54059cbbae"},"response":null} []
2020-02-28T18:14:09Z ERROR admin.API.Restore {"duration":0.277950744,"error":"error creating commit: parent commit a99f85b89bd64df186dc207e7fd20de3 has not been finished","request":null} []
```

Inspecting the job that gave error `commit f7a48dd`..

```
Commit: extraction_exporter@f7a48dde6aae4ddfb9374a54059cbbae
Original Branch: master
Parent: a99f85b89bd64df186dc207e7fd20de3
Started: 4 weeks ago
Finished: 4 weeks ago
Size: 0B
Provenance:  __spec__@ced850264e9a4c6b9f40e5bf7358411c (extraction)  __spec__@b400e6df43aa408ca43d68bf3cafd1ce (extraction_exporter)  extraction@b26887a935484b718bc82004872a2fc3 (master)  extraction_morning@187d036de83042358e6654bf80ff86c2 (master)
```

Inspecting the parent commit `a99f85b`..
```
Commit: extraction_exporter@a99f85b89bd64df186dc207e7fd20de3
Original Branch: master
Started: 4 weeks ago
Finished: 4 weeks ago
Size: 0B
Provenance:  __spec__@80fe63de354a4166bb6fff0710e288bf (extraction)  __spec__@b400e6df43aa408ca43d68bf3cafd1ce (extraction_exporter)  extraction@1959138c183f4cd4b3df36ca6013759f (master)  extraction_morning@187d036de83042358e6654bf80ff86c2 (master)
```

The refered commit is actually finished 4 weeks ago.

**What you expected to happen?**:

Restore successfully or fix the commit problem.

**How to reproduce it (as minimally and precisely as possible)?**:

N/A

**Anything else we need to know?**:

Tried fsck and garbage-collect to no avail.

**Environment?**:
- Kubernetes version (use `kubectl version`):
Server Version: version.Info{Major:"1", Minor:"12+", GitVersion:"v1.12.10-eks-aae39f", GitCommit:"aae39f4697508697bf16c0de4a5687d464f4da81", GitTreeState:"clean", BuildDate:"2019-12-23T08:19:12Z", GoVersion:"go1.12.9", Compiler:"gc", Platform:"linux/amd64"}

- Pachyderm CLI and pachd server version (use `pachctl version`):
COMPONENT VERSION
pachctl 1.9.12
original cluster: pachd 1.9.11
new cluster: pachd 1.9.12

- Cloud provider (e.g. aws, azure, gke) or local deployment (e.g. minikube vs dockerized k8s):
AWS

- OS (e.g. from /etc/os-release):
- Others:

### PR DESCRIPTION
**Pull request by brycemcanally**
This PR fixes some issues in the extract/restore process. The fixes have to do with the conversion of 1.9.x data structures to 1.10. Several fields were left out in the conversion file:

CreateJobRequest: DataFailed
SecretMount: Key, MountPath, EnvVar
Transform: ErrCmd, ErrStdin
PFSInput: JoinOn
Service: Type
BuildCommitRequest: Provenance (we are extracting / restoring output commits now), Trees, Datums, SizeBytes
CreatePipelineRequest: Spout, PodPatch, SpecCommit (this was causing the extra commit issue I was seeing).

@jdoliner @msteffen please provide feedback if any of these fields should be left out / changed.

Fixes #4661

### CODE REVIEW

File: src/server/admin/server/convert1_9.go [hunk 1]
```diff
@@ -203,19 +203,30 @@ func convert1_9Secrets(secrets []*pps1_9.Secret) []*pps.SecretMount {
...
+
 func convert1_9Transform(t *pps1_9.Transform) *pps.Transform {
 	if t == nil {
 		return nil
 	}
 	return &pps.Transform{
 		Image:            t.Image,
 		Cmd:              t.Cmd,
-		ErrCmd:           nil,
+		ErrCmd:           copyStringSlice(t.ErrCmd),
```
Comment (jdoliner) [inline src/server/admin/server/convert1_9.go]: Is there a reason we need to do a copy here but not on `t.Cmd` it seems like they should be being handled the same. My guess is you don't need to copy here but I might be wrong.

File: src/server/admin/server/convert1_9.go [hunk 2]
```diff
@@ -203,19 +203,30 @@ func convert1_9Secrets(secrets []*pps1_9.Secret) []*pps.SecretMount {
...
 }
 
+func copyStringSlice(ss []string) []string {
+	if ss == nil {
+		return nil
+	}
+	result := make([]string, 0, len(ss))
+	for _, s := range ss {
+		result = append(result, s)
+	}
```
Comment (jdoliner) [inline src/server/admin/server/convert1_9.go]:
```suggestion
	result := make([]string, len(ss))
	copy(result, ss)
```

I think this is a little terser and in theory a little more efficient (not that it really matters here.)

File: src/server/admin/server/convert1_9.go (hunk 1 above)
Comment (brycemcanally) [reply src/server/admin/server/convert1_9.go]: Yeah, I guess you don't. I was thinking the conversion functions might need to do a deep copy, but I guess they are just there so particular conversions can be applied in one place.

File: src/server/admin/server/convert1_9.go (hunk 2 above)
Comment (brycemcanally) [reply src/server/admin/server/convert1_9.go]: I decided to follow the same pattern found throughout this file to be consistent.

**Pull request merged**